### DISCUSSION TITLE
Questions about how to use veramo cli Ⅱ

### DISCUSSION THREAD

**Discussion by cbw1030** [decentralized-identity/veramo#663] [Q&A]
This is the writer who posted the previous post.
I'm asking you a question again just in case you didn't read the comments.

Thank you for your answer.
Run with 'veramo server' command and verify good connection with 'localhost:3332/api-docs'.
We also checked the source code for 'localhost:3332/open-api.json'.
But there are still some things I don't understand, so I'm leaving a comment.

1. Attempting to connect to 'localhost:3332/messaging' results in 'Cannot GET/messaging' error. Why is that?
Attempting to connect to 'localhost:3332/agent' results in an 'Unauthorized' error.
I ran it on the local server, but why does the above error occur when I connect with '/agent'?

2. When you access '/api-docs', the 'Authorize' button exists.
I'd like to know what the button means. It's my first time using Swagger UI.
If you set the value and refresh it, it will be released.

3. As you answered, I'm looking for schema in /open-api.json, but I'm not sure what it means.
Can you give me an example, assuming you enter schema in 'https://expore.veramo.dev/connect'?

4. Enter the 'npx agent-expore service' command to access localhost:5000.
I wonder the difference between the localhost:5000 site and the https://expore.veramo.dev/connect site.

5. As you have answered, you can access as follows at node.js by referring to ngrok.io.
localhost:3332/api-docs -> https://15d73b196ada.ngrok.io/api-docs/
But I don't know what to do afterwards.
Can I communicate with my local DID, Credential and https://expore.veramo.dev/connect here?

Thank you for reading the long question.

**Comment by mirceanis** [comment]
1. Each of the exposed endpoints has a purpose and they are not meant to be human readable (except for `/api-docs`)
The `/messaging` endpoint listens to HTTP POST which is why it fails on GET. It can be used by agents that want to talk to each other using [`DIDComm`](https://github.com/decentralized-identity/didcomm-messaging)
The `/agent` endpoint requires an `Authorization` header with the proper API key (which you can configure in `agent.yml`).
This is used by frontend applications that need to execute agent methods remotely.

2. The `Authorize` button allows you to paste the API key (that you can configure in `agent.yml`) so that you can execute agent methods from the web interface.

3. The public URL of the schema is what needs to be pasted into https://expore.veramo.dev/connect
In your case, it would be https://15d73b196ada.ngrok.io/open-api.json
Also, be advised that the free `ngrok` hostname changes every time you restart ngrok.

4. There should be no difference. It's the same code that is hosted on explore.veramo.dev

5. You can use the Authorize button to gain the ability to call agent methods. This is not really `communicating with your local DID`, but rather communicating with the agent that manages that DID.
